src/DIRAC/WorkloadManagementSystem/Executor/Scouting.py

```diff
@@ -0,0 +1,131 @@
+""" Executor to set status "Scouting" for a main job which has scout jobs
+"""
+
+from DIRAC import S_OK, S_ERROR
+
+from DIRAC.WorkloadManagementSystem.Executor.Base.OptimizerExecutor import OptimizerExecutor
+from DIRAC.WorkloadManagementSystem.Client import JobStatus
+from DIRAC.ResourceStatusSystem.Client.SiteStatus import SiteStatus
+from DIRAC.WorkloadManagementSystem.Client.JobStateUpdateClient import JobStateUpdateClient
+
+
+class Scouting(OptimizerExecutor):
+    """
+      The specific Optimizer must provide the following methods:
+      - optimizeJob() - the main method called for each job
+      and it can provide:
+      - initializeOptimizer() before each execution cycle
+    """
+
+    @classmethod
+    def initializeOptimizer(cls):
+        """ Initialization of the optimizer.
+        """
+        cls.__jobDB = JobStateUpdateClient()
+        return S_OK()
+
+    def optimizeJob(self, jid, jobState):
+        self.jobLog.info('Getting scoutparams from JobParameters')
+
+        result = self.__jobDB.getJobParameters(jid, ['ScoutFlag', 'ScoutID'])
+        if not result['OK']:
+            return result
+
+        rCounter = 0
+        if result['Value']:
+            scoutparams = result['Value'].get(jid)
+            self.jobLog.info('scoutparams: %s' % scoutparams)
+            if not scoutparams:
+                self.jobLog.info('Skipping optimizer, since scoutparams are abnormal')
+                return self.setNextOptimizer(jobState)
+
+            scoutID, scoutFlag = self.__getIDandFlag(scoutparams)
```

Comment:
```suggestion
            scoutID, scoutFlag = scoutparams.get('ScoutID'), scoutparams.get('ScoutFlag')
```
and you can remove `self.__getIDandFlag`